src/ring_theory/subsemiring.lean

```diff
@@ -369,6 +420,45 @@ lemma closure_eq_of_le {s : set R} {t : subsemiring R} (h₁ : s ⊆ t) (h₂ :
   closure s = t :=
 le_antisymm (closure_le.2 h₁) h₂
 
+end subsemiring
+
+namespace submonoid
+
+lemma to_subsemiring_coe :
+  (M.to_subsemiring : set R) = add_submonoid.closure (M : set R) := rfl
+
+end submonoid
+
+namespace subsemiring
+
+lemma to_subsemiring_to_add_submonoid :
+  M.to_subsemiring.to_add_submonoid = add_submonoid.closure (M : set R) := rfl
+
+/-- The `subsemiring.closure` of a multiplicative submonoid coincides with the `subsemiring`
+generated by the submonoid itself. -/
+lemma closure_eq_submonoid_to_subsemiring : subsemiring.closure (M : set R) = M.to_subsemiring :=
+begin
```

Comment:
I think this lemma might be easier to find stated in reverse (and inside the submonoid namespace):
```suggestion
lemma to_subsemiring_eq_closure : M.to_subsemiring = subsemiring.closure (M : set R) :=
begin
```
If you can come up with a useful phrasing for the docstring feel free to keep it, but I think the statement stands without one too,

Reply:
Ok, I agree: with the new layout, the doc-string would be saying essentially the same that the formal statement already asserts.